File: Cano-Rodriguez-Claudio/Procesos/Java/thread.java
public class ThreadExample {

    public static void main(String[] args) {
        // Child thread
        Thread childThread = new Thread(() -> {
            while (true) {
                System.out.println("\t\t\tH=" + ProcessHandle.current().pid());
            }
        });

        childThread.start(); // Start the child thread

        // Main thread
        while (true) {
            System.out.println("\tP=" + ProcessHandle.current().pid());
        }
    }
}

File: Cano-Rodriguez-Claudio/Procesos/C/thread.c
#include <stdio.h>      // For printf
#include <pthread.h>    // For pthread_create
#include <unistd.h>     // For getpid()

// Function executed by the child thread
void thread(void) {
    while (1) {
        printf("\t\t\tH=%d\n", getpid()); // Print PID with indentation
    }
}

int main(int argc, char *argv[]) {
    int pid;
    pthread_t tid; // Thread ID

    // Create a new thread
    pid = pthread_create(&tid, NULL, (void *)thread, NULL);
    if (pid < 0) {
        printf("ERR: no se creo el hilo\n"); // Error if thread creation fails
        return 1;
    }

    // Main thread continuously prints its PID
    while (1) {
        printf("\tP=%d\n", getpid()); // Print PID with single tab
    }
}
